src/structures/CategoryChannel.js

```diff
@@ -17,6 +17,36 @@ class CategoryChannel extends GuildChannel {
   get children() {
     return this.guild.channels.filter(c => c.parentID === this.id);
   }
+  /**
+   * Creates a new channel with this category as its parent.
+   * @param {string} name The name of the new channel
+   * @param {ChannelData} [options] Options for the new channel
+   * @returns {Promise<CategoryChannel|TextChannel|VoiceChannel>}
+   * @example
+   * // Create a new text channel
+   * category.createChannel('new-general')
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Create a new channel with permission overwrites
+   * category.createChannel('new-restricted', {
+   *   permissionOverwrites: [{
+   *     id: guild.id,
+   *     deny: ['MANAGE_MESSAGES'],
+   *     allow: ['SEND_MESSAGES']
+   *   }]
+   * })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  createChannel(name, options) {
+    if (!options) {
+      options = { parent: this.id };
```

Comment:
I think that that requires a bit more logic, as CategoryChannel cannot hold another CategoryChannel as its child.